apps/store/src/blocks/TabsBlock.tsx

```diff
@@ -0,0 +1,25 @@
+import { SbBlokData, StoryblokComponent, storyblokEditable } from '@storyblok/react'
+import * as Tabs from '@/components/ProductPage/Tabs'
+import { SbBaseBlockProps } from '@/services/storyblok/storyblok'
+import { TabBlockFields } from './blocks.types'
+
+type TabsBlockProps = SbBaseBlockProps<{
+  items: Array<SbBlokData & TabBlockFields>
+}>
+
+export const TabsBlock = ({ blok }: TabsBlockProps) => {
+  return (
+    <Tabs.Tabs defaultValue={blok.items[0]?._uid} {...storyblokEditable(blok)}>
+      <Tabs.TabsList>
+        {blok.items.map((item) => (
+          <Tabs.TabsTrigger key={item._uid} value={item._uid}>
+            {item.title}
+          </Tabs.TabsTrigger>
+        ))}
+      </Tabs.TabsList>
+      {blok.items.map((nestedBlock) => (
+        <StoryblokComponent key={nestedBlock._uid} blok={nestedBlock} />
+      ))}
```

Comment:
[Discussion]
Maybe it's just the way _Storyblok_ works but I get the sense that this solution might be too implicit and therefore hard to understand. The issue here is that, even tho we can define in _Storyblok_ that only `TabBlock` should be accepted as children of `TabsBlock`, that relationship is not well expressed in the code. E.g: here we're rendering `<StoryblokComponent />` which can be any block, but at the end it will became a `TabBlock`.

My first thought would be changed it for `<TabBlock>` and don't register a component for it inside `storybook` service.

```suggestion
      {blok.items.map((nestedBlock) => (
        <TabBlok key={nestedBlock._uid} blok={nestedBlock} />
      ))}
```
But to be honest I don't thing I've a solution for it. I've just thought it would be a good thing to discuss and I was hoping to see other opinions about it :)

Reply:
@guilhermespopolin very good point, I'm also not very happy with this setup.

@alebedev and I had similar discussion around the accordion. But here it becomes even more obvious.

I'm leaning towards your idea if just using the `TabBlock` inside the `TabsBlock` instead of using `StoryblokComponent` - you made a good case!

Reply:
Thanks :)
I'm glad I could help